lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
+      fail HawkularException.new('Invalid credentials', http_code) if f.instance_of? RestClient::Unauthorized
```

Comment:
@pilhuhn understood the limitations of wildfly. @rubenvp8510 I am less familiar which errors return as json and which is html, so whatever you two find the best approach is :+1: from me.